Issue#189 add newtab user data

FrancescoSTL opened the pull request:
<img width="1194" alt="screen shot 2016-06-28 at 3 43 30 pm" src="https://cloud.githubusercontent.com/assets/9794516/16434876/1bed79ca-3d47-11e6-9165-c1502ce4d4d9.png">

Issue ref: #189 

We are removing the footer when a user has sync enabled, because if they do, the undo button will not work.

r? @schalkneethling

schalkneethling commented on data/css/about-newtab.css:
```diff
@@ -0,0 +1,13 @@
+}
+
+.importBody {
+	font-size: 15px;
```
Nit: Consistent indentation of 4 spaces

schalkneethling commented on index.js:
```diff
@@ -743,6 +747,38 @@ function modifyFirstrun() {
+        onAttach: function(worker) {
+            // constructs uri to snippet content
+            var headContentURL = './tmpl/about-newtab-header.html';
+            var footContentURL = './tmpl/about-newtab-footer.html';
```
Nit: For consistency, let's call this `footerContentURL`

schalkneethling commented on data/tmpl/about-newtab-footer.html:
```diff
@@ -0,0 +1,3 @@
+<div class="autoImport">
```
Nit: Use `section` for consistency and semantics

schalkneethling commented on data/tmpl/about-newtab-header.html:
```diff
@@ -0,0 +1,9 @@
+<div class="autoImport">
```
Nit: Use `section` for consistency and semantics

schalkneethling commented on data/tmpl/about-newtab-header.html:
```diff
@@ -0,0 +1,9 @@
+  <header>
+    <img src="%url" width="65" height="40" class="moving-truck" alt="Image of moving truck" />
+  </header>
+  <div>
```
change this to `main`

schalkneethling commented on data/tmpl/about-newtab-header.html:
```diff
@@ -0,0 +1,9 @@
+  </header>
+  <div>
+    <h1>We've got your back.</h1>
+    <p class="importBody">Firefox has already imported your preferences from other browsers you've used in the past.</p>
```
Nit: For consistency name classes as `import-body`

schalkneethling commented on data/css/about-newtab.css:
```diff
@@ -0,0 +1,13 @@
+.autoImport {
```
Nit: For consistency change to `auto-import`

schalkneethling commented:
Couple of nits and suggestions. Looking good overall. Needs a rebase. r+ w/c 📑

chrismore commented:
Mentioned in #189 too:


@FrancescoSTL the undo functionality has landed, it won't be functional on our test builds, but we need to figure out how to call it from the undo link at the bottom of about:newtab.

Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1271799

FrancescoSTL commented:
Fixed various, added rebase, added preliminary undo logic (commented, for now). Seeing as we can't test the undo logic yet, I added what I believe is the correct solution, however, we can update upon update of the bugzilla thread: https://bugzilla.mozilla.org/show_bug.cgi?id=1271799#c13

r? @schalkneethling

schalkneethling commented on data/css/about-newtab.css:
```diff
@@ -0,0 +1,13 @@
+}
+
+.import-body {
+    font-size: 15px;
```
Nit: use `rem` for font sizing. I reckon is probably `1rem` or perhaps `.9rem`

schalkneethling commented on data/js/about-newtab.js:
```diff
@@ -0,0 +1,20 @@
+
+undoMigrate.addEventListener('click', function() {
+    addon.port.emit('undoMigrate');
+});
```
No empty line at end of the file.

schalkneethling commented:
Two small nits, other than that r+ w/c 📰

FrancescoSTL commented:
Done and done. Thanks @schalkneethling 🏈
